cmd/tikv-ctl/src/main.rs

```diff
@@ -1050,867 +1051,661 @@ fn warning_prompt(message: &str) -> bool {
+    CompactCluster {
+        #[structopt(
+            short = "d",
+            default_value = "kv",
+            possible_values = &["kv", "raft"],
+        )]
+        /// The db to use
+        db: String,
+
+        #[structopt(
+            short = "c",
+            use_delimiter = true,
+            require_delimiter = true,
+            value_delimiter = ",",
+            default_value = CF_DEFAULT,
+            possible_values = &["default", "lock", "write"],
+        )]
+        /// Column family names, for kv db, combine from default/lock/write; for raft db, can only be default
+        cf: Vec<String>,
+
+        #[structopt(
+            short = "f",
+            long,
+            help = RAW_KEY_HINT,
+        )]
+        from: Option<String>,
+
+        #[structopt(
+            short = "t",
+            long,
+            help = RAW_KEY_HINT,
+        )]
+        to: Option<String>,
+
+        #[structopt(short = "n", long, default_value = "8")]
+        /// Number of threads in one compaction
+        threads: u32,
+
+        #[structopt(
+            short = "b",
+            long,
+            default_value = "default",
+            possible_values = &["skip", "force", "default"],
+        )]
+        /// How to compact the bottommost level
+        bottommost: String,
+    },
+    /// Show region properties
+    RegionProperties {
+        #[structopt(short = "r")]
+        /// The target region id
+        region: u64,
+    },
+    /// Show range properties
+    RangeProperties {
+        #[structopt(long, default_value = "")]
+        /// hex start key
+        start: String,
+
+        #[structopt(long, default_value = "")]
+        /// hex end key
+        end: String,
+    },
+    /// Split the region
+    SplitRegion {
+        #[structopt(short = "r")]
+        /// The target region id
+        region: u64,
+
+        #[structopt(short = "k")]
+        /// The key to split it, in unencoded escaped format
+        key: String,
+    },
+    /// Inject failures to TiKV and recovery
+    Fail {
+        #[structopt(subcommand)]
+        cmd: FailCmd,
+    },
+    /// Print the store id
+    Store {},
+    /// Print the cluster id
+    Cluster {},
+    /// Decrypt an encrypted file
+    DecryptFile {
+        #[structopt(long)]
+        /// input file path
+        file: String,
+
+        #[structopt(long)]
+        /// output file path
+        out_file: String,
+    },
+    /// Dump encryption metadata
+    EncryptionMeta {
+        #[structopt(subcommand)]
+        cmd: EncryptionMetaCmd,
+    },
+    /// Print bad ssts related infos
+    BadSsts {
+        #[structopt(long)]
+        /// db directory.
+        db: String,
+
+        #[structopt(long)]
+        /// specify manifest, if not set, it will look up manifest file in db path
+        manifest: Option<String>,
+
+        #[structopt(long, value_delimiter = ",")]
+        /// PD endpoints
+        pd: String,
```

Comment:
Should this be `Vec<String>` instead?

Reply:
The current code in master is confusing.

On one side, `value_delimiter` is [presented](https://github.com/tikv/tikv/blob/master/cmd/tikv-ctl/src/main.rs#L1902), and the [help text](https://github.com/tikv/tikv/blob/master/cmd/tikv-ctl/src/main.rs#L1903) indicates that it accepts multiple arguments. On the other side, `.multiple(true)` is *not* presented, and the argument is [used](https://github.com/tikv/tikv/blob/master/cmd/tikv-ctl/src/main.rs#L1903) as a single value. Thus in master, this leads a strange behaviour that it do accept multiple arguments, but only the first one is used. And this PR have the same help information and usage logic as master, with only difference that multiple arguments will results a panic, the behaviour which I think is not worse.

This PR is indented to have as few behaviour change as possible. So my suggestion is to open a new issue to address this, and keep the status quo in this PR.

Reply:
The current code in master is confusing.

On one side, `value_delimiter` is [presented](https://github.com/tikv/tikv/blob/master/cmd/tikv-ctl/src/main.rs#L1902), and the [help text](https://github.com/tikv/tikv/blob/master/cmd/tikv-ctl/src/main.rs#L1903) indicates that it accepts multiple arguments. On the other side, `.multiple(true)` is *not* presented, and the argument is [used](https://github.com/tikv/tikv/blob/master/cmd/tikv-ctl/src/main.rs#L1903) as a single value. Thus in master, this leads a strange behaviour that it do accept multiple arguments, but only the first one is used. And this PR have the same help information and usage logic as master, with only difference that multiple arguments will results a panic, the behaviour which I think is not worse.

This PR is indented to have as few behaviour change as possible. So my suggestion is to open a new issue to address this, and keep the status quo in this PR.